### PR TITLE
37 save data between views   use scenarios

### PR DESCRIPTION
**Pull request by amynickolls**
Created both Session and Saved Scenarios. Session scenarios are how we will transfer data with the user between views. In this implementation it saves the historic data filters, but it will also save any adjustments or prediction inputs as we develop those pages.

Session scenarios are created/accessed through a router_handler that redirects to the relevant URL.

Dashboard page and relevant code has been deleted as this is not in the designs and was complaining re: the updated scenario model.

### CODE REVIEW

**Review by tab1tha — APPROVED** [review]
- When filters such as `Placement_type=Residential` and `Age=10 to 16` are added to the historic data tab and then the user clicks away from the tab, the filters remain in place when the user comes back to the tab. This isn't possible on the main branch. 
So the changes in this pull request successfully make filters persist.

**Comment by tab1tha** [comment]
Other notes 
**About this pull request.**
- I wasn't sure where to click in order to save a scenario.
- When the user filters the historic view to only see 10-16 year-olds in residential placement. Should the plots in the prediction graph only be those which are relevant to the filters in the historic data tab?. Now, the prediction graph seems to remain unchanged irrespective of the historic data filters.

**In general**
-  When I `Empty cache and hard reload` the browser page (to delete cookies), the application keeps me logged on. Flagging this because it might be expected that the user is logged out when cookies are cleared. It is also possible that my cookies are not being cleared in the way that I think so feel free to ignore this if it works okay on your end.
- It could be nice if the scenario view page had a button from where the user could run the scenario. (this is a screenshot from the scenarios page in the main branch. Didn't get to try out the one in this branch since I couldn't successfully save a scenario so feel free to ignore this if that feature has already been built) 
image

**Comment by amynickolls** [comment]
Thanks Tambe! All expected for now. Both saving and loading saved scenarios are coming in an issue down the line. Likewise the historic filters will affect the prediction page but this is yet to be implemented.

As for the log in issue might be that you need to close page then clear cache then return?

**Review by cyramic — COMMENTED** [review]
Happy to approve, but a few questions first
* Why are the print() statements there?  Would they be better to remove or to switch to logging statements?
* Is deleting Scenario and adding SavedScenario, do we need to worry about the migration of data?

File: dm_regional_app/management/commands/sandbox.py [hunk 1]
```diff
@@ -31,7 +33,16 @@ def handle(self, *args, **kwargs):
         pop = PopulationStats(dc.enriched_view, config)
 
         data = dc.enriched_view
-        print(data.loc[data.UASC == True])
+        # print(data.loc[data.UASC == True])
 
         # print(pop.stock)
         # print(dc.enriched_view)
+        session_scenario = get_object_or_404(SessionScenario, pk=1)
+        print(session_scenario.historic_filters)
```
[the remark this answers is not in the document]
Comment (amynickolls) [reply dm_regional_app/management/commands/sandbox.py:41]: The print statements in sandbox are only for my own testing/debugging, the whole thing will be deleted once developed. Is there a better way to do that? I want to keep the commented code just for returning to.

File: dm_regional_app/migrations/0003_savedscenario_sessionscenario_delete_scenario.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
+# Generated by Django 4.2.10 on 2024-04-24 15:30
```
[the remark this answers is not in the document]
Comment (amynickolls) [reply dm_regional_app/migrations/0003_savedscenario_sessionscenario_delete_scenario.py:1]: Not live no!

**Pull request merged**